R/tm_g_gh_spaghettiplot.R

```diff
@@ -400,7 +397,14 @@ srv_g_spaghettiplot <- function(id,
 
     # update sliders for axes taking constraints into account
     yrange_slider <- toggle_slider_server("yrange_scale")
-    keep_range_slider_updated(session, input, yrange_slider$update_state, "yaxis_var", "xaxis_param", anl_q)
+    observe({
+      yrange_slider <- keep_slider_state_updated(
+        state = yrange_slider,
+        varname = input$yaxis_var,
+        paramname = input$xaxis_param,
+        ANL = anl_q()$ANL
+      )
+    })
```

Comment:
IMO this should be done in other way. Call `keep_slider_state_updated` directly and move observe to the function. WDYT?
```suggestion
    keep_slider_state_updated(
      state = yrange_slider,
      varname = reactive(input$yaxis_var),
      paramname = reactive(input$xaxis_param),
      ANL = reactive(anl_q()$ANL)
    )
```

Reply:
Good catch that the assignment is not needed here.
But, moving reactive code into a function is always challenging to maintain. I'll check if we can move the observe inside the slider module.

Reply:
I like the current way of split (modules + update logic in a function) because we don't restrict the toggle slider shiny module with the update logic, which might change for different modules. For example the module `tm_g_gh_lineplot.R` does not use the `keep_slider_state_updated`. So, name is misleading by suggesting this is a generic function to keep the slider state updated. I'll rename it to something specific to the range setting logic that it does.

Reply:
IMO it is a bad way. In `toggle_slider_server` `state$data_range` is created but for `keep_slider_updated`. `data_range` should be a property of `keep_slider_updated` and `keep_slider_updated` should update `state$value` directly when range changes. 

I've checked lineplot, I see the difference but the lineplot is not an argument for `keep_slider_updated` if it does not use it. This whole observe/update thing is wrong, we can discuss this on the call.